### ISSUE TITLE
immortal phare

### ISSUE THREAD

**Issue by nicolasaunai** [PHAREHUB/PHARE#227]
can't Ctrl-C phare-exe once started...
At the end of the execution I get

```
Exception ignored in: <module 'threading' from '/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py'>
Traceback (most recent call last):
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 1274, in _shutdown
    def _shutdown():
```

**Comment by Dekken** [comment]
how do you run phare.exe? running it with no args seg faults.
While we're at this should we move the phare.exe binary to another location?

$BUILD/bin
rather than
$BUILD/src/phare

**Comment by nicolasaunai** [comment]
```
./phare-exe phare_init.py
```

phare_init.py should be copied in the phare-exe dir by cmake

**Comment by Dekken** [comment]
> phare_init.py should be copied in the phare-exe dir by cmake

I don't think this is the case on master.

**Comment by Dekken** [comment]
it's here https://github.com/PHAREHUB/PHARE/pull/115/files#diff-5ce1f5e96daeccf6ec83e1eb3f551e28R12

**Comment by nicolasaunai** [comment]
yeah sorry, I meant of course on advancement because otherwise phare-exe does pretty much nothing :/

**Comment by nicolasaunai** [comment]
it's now on master

**Comment by Dekken** [comment]
I can reproduce this

**Comment by Dekken** [comment]
we may want to consider something like this to kill the global dict

```https://github.com/Dekken/maiken/blob/master/server.cpp#L55```

**Comment by Dekken** [comment]
a possible solution might be to put PhareDict.stop() in ~PythonDataProvider()

**Comment by nicolasaunai** [comment]
can be closed no?

**Comment by Dekken** [comment]
it might still be an issue

**Comment by PhilipDeegan** [comment]
possible work around is to kill the PythonDataProvider after "simulator->initialize();"

**Comment by PhilipDeegan** [comment]
[suggested solution ](https://gist.github.com/PhilipDeegan/f571a42ce6dea1de639ce59aba6019b2)
